modules/layers/src/geojson-layer/geojson.js

```diff
@@ -81,45 +83,69 @@ export function separateGeojsonFeatures(features) {
     } = feature;
     checkCoordinates(type, coordinates);
 
+    const __deckMetadata = {
```

Comment:
The naming here is strange. `geojsonPickingData`?

Reply:
I thought `__deck` could be a good prefix to ensure it doesn't clash with any other custom properties users may have on their geojson

Reply:
Standard GeoJSON features should not contain any field other than `geometry` and `properties`.

Reply:
The spec says users MAY provide additional properties, but with caution.
https://tools.ietf.org/html/rfc7946#section-6

Reply:
Nit: It may be too hard to do, and I don't want to block this PR, but I think would prefer if it was possible to keep this information in a separate structure, i.e. not adding data to the main geojson. I haven't thought deeply about it so not sure how easy it is to do that.

If we do need to store this inline in the geojson, maybe change the name to make it clear it is related to picking: `_deckPickInfo`?

I also feel we would benefit from some picking tests to make sure this doesn't break as we keep refactoring this already somewhat complex code.

Reply:
Regarding storing in a separate structure: That's actually already happening. `separateGeojsonFeatures` returns an object with a separate structure and doesn't modify the features passsed in.

Regarding the name: I realized it actually doesn't matter the name. We write that name to the separate structure, then wrap the source feature inside it. Then when picking info is returned, we unwrap that. So the name only exists within `GeoJsonLayer`. The user will get the exact same feature they handed in the props in the returned pick info. I'll just name it `deckPickingInfo` so that it is clear.